testing/integration/agent_long_test.go

```
@@ -0,0 +1,294 @@
+		handlesReg.SetVar(0, "time")
+
+		runner.T().Logf("created handle watcher for %s (%d)", comp.Name, pid)
+		handles = append(handles, processWatcher{handle: handle, pid: int(pid), name: comp.Name, regHandles: handlesReg})
+	}
+
+	testDuration, err := time.ParseDuration(testRuntime)
+	require.NoError(runner.T(), err)
+
+	timer := time.NewTimer(testDuration)
+	defer timer.Stop()
+
+	// time to perform a health check
+	ticker := time.NewTicker(time.Second * 10)
+	defer ticker.Stop()
+
+	done := false
+	start := time.Now()
+	for !done {
+		select {
+		case <-timer.C:
+			done = true
+		case <-ticker.C:
+			err := runner.agentFixture.IsHealthy(ctx)
+			require.NoError(runner.T(), err)
+			// for each running process, collect memory and handles
+			for _, handle := range handles {
+
+				ohc, ok := handle.handle.(types.OpenHandleCounter)
+				if ok {
+					handleCount, err := ohc.OpenHandleCount()
+					require.NoError(runner.T(), err)
+					handle.regHandles.Train(regression.DataPoint(float64(handleCount), []float64{time.Since(start).Seconds()}))
+				}
+
+			}
+		}
+	}
+
+	// we're measuring the handle usage as y=mx+b
+	// if the slope is increasing above a certain rate, fail the test
+	// A number of factors can change the slope during a test; shortened runtime (lots of handles allocated in the first few seconds, producing an upward slope),
+	// filebeat trying to open a large number of log files, etc
+	handleSlopeFailure := 0.1
+
+	for _, handle := range handles {
+		err = handle.regHandles.Run()
+		require.NoError(runner.T(), err)
+
+		runner.T().Logf("=============================== %s (%d)", handle.name, handle.pid)
+		runner.T().Logf("handle formula: %v", handle.regHandles.Formula)
+		coeffs := handle.regHandles.GetCoeffs()
+		handleSlope := coeffs[1]
+		// This is a hack to deal with the fact that we'll pick up zombie processes that seem to happen when agent restarts
+		if math.IsNaN(handleSlope) {
+			continue
+		}
```

Agent shouldn't be restarting when the test starts, is this actually finding a bug?

This also has the potential to hide failures in the test. What if we skipped the `LessOrEqual` test for every slope calculation. You need to guard against that.